CuBeanzQueryable.sol

```diff
@@ -0,0 +1,251 @@
+        if(rOneMapping[msg.sender] + amount > maxRoundOneMints) revert MaxMints();
+         rOneMapping[msg.sender]+=amount;
+         roundOneMints += amount;
+         _mint(msg.sender,amount);
+    }
+
+    function roundTwoMint(uint8 amount) external  {
+        if(!roundTwoLive) revert SaleNotStarted();
+        if(msg.sender != tx.origin) revert ContractMint();
+        if(roundTwoMints + amount > roundTwoSupply) revert RoundSoldOut();
+        if(rTwoMapping[msg.sender] + amount > maxRoundTwoMints) revert MaxMints();
+        rTwoMapping[msg.sender] += amount;
+        roundTwoMints += uint16(amount);
+        _mint(msg.sender,amount);
+    }
+
+
+     function ryoMint(uint8 amount) external{
+         if(!burnMintActive) revert SaleNotStarted();
+         ryoToken.transferFrom(msg.sender,address(this), cubePrice * amount);
+         _mint(msg.sender,amount);
+     }
+
+     function attackBean(uint16 tokenId) external{
+        if(msg.sender != ownerOf(tokenId)) revert NotOwner();
+         require(numAttacks[tokenId] < maxAttacks);
+         numAttacks[tokenId]++;
+         emit ATTACK(numAttacks[tokenId],tokenId);
+     }
+
+
+    function burnCube(uint16 tokenId) external {
+        if(msg.sender != ownerOf(tokenId)) revert NotOwner();
+        ryoToken.mint(msg.sender, burnRate);
+        _burn(tokenId);
+    }
+
+
+
+    function burnBatchCubes(uint16[] calldata tokenIds) external{
+        for(uint16 i; i<tokenIds.length;i++){
+            if(msg.sender != ownerOf(tokenIds[i])) revert NotOwner();
+            _burn(tokenIds[i]);
+        }
+        ryoToken.mint(msg.sender, burnRate * tokenIds.length);
+    }
+
+
+    //SETTERS
+    function setMaxAttacks(uint8 amountAttacks) external onlyOwner {
+        require(amountAttacks > maxAttacks);
+        maxAttacks = amountAttacks;
+    }
+
+    function setRyoToken(address _address) external onlyOwner{
+        ryoToken =  IERC20(_address);
+    }
+
+    function setRoundMaxSupply(uint8 roundNumber ,uint16 newSupply) external onlyOwner {
+        if(roundNumber ==1){
+            roundOneSupply = newSupply;
+        }
+        if(roundNumber == 2){
+            roundTwoSupply = newSupply;
+        }
+    }
+
+    function setRevealed(bool status) public onlyOwner {
+        revealed = status;
+    }
+
+    function setNotRevealedURI(string memory _notRevealedURI) public onlyOwner {
+        notRevealedUri = _notRevealedURI;
+    }
+
+    function setBaseURI(string memory _newBaseURI) public onlyOwner {
+        baseURI = _newBaseURI;
+    }
+
+   
+    function setUriSuffix(string memory _newSuffix) external onlyOwner{
+        uriSuffix = _newSuffix;
+    }
+
+      function setSaleStarted(uint8 roundNumber, bool status) external onlyOwner{
+        if(roundNumber == 1){
+            roundOneLive = status;   
+        }
+        if(roundNumber == 2){
+            roundTwoLive = status;
+        }
+        if(roundNumber == 3){
+            burnMintActive = status;
+        }
+    }
```

Comment:
Must round 1, round 2, and burn mint active be mutually exclusive?

If so, consider using a single uint8 `saleState` to represent the state.